test/integration/users.test.js

```diff
@@ -11,14 +11,91 @@ const cleanDb = require("../utils/cleanDb");
 const userData = require("../fixtures/user/user")();
 const profileDiffData = require("../fixtures/profileDiffs/profileDiffs")();
 const superUser = userData[4];
+const searchParamValues = require("../fixtures/user/search")();
 
 const config = require("config");
 const joinData = require("../fixtures/user/join");
-const { addJoinData } = require("../../models/users");
+const { addJoinData, addOrUpdate } = require("../../models/users");
 const cookieName = config.get("userToken.cookieName");
 
 chai.use(chaiHttp);
 
+const userSearchParamsData = [
+  {
+    username: "ankur",
+    first_name: "Ankur",
+    last_name: "Narkhede",
+    yoe: 0,
+    img: "./img.png",
+    linkedin_id: "ankurnarkhede",
+    github_id: "ankur1234",
+    github_display_name: "ankur-xyz",
+    phone: "1234567890",
+    email: "abc@gmail.com",
+  },
+  {
+    username: "23ankur",
+    first_name: "Ankur",
+    last_name: "Narkhede",
+    yoe: 0,
+    img: "./img.png",
+    linkedin_id: "ankurnarkhede",
+    github_id: "ankur1234",
+    github_display_name: "ankur-xyz",
+    phone: "1234567890",
+    email: "abc@gmail.com",
+  },
+];
+
+searchParamValues.forEach(function (item, i) {
+  describe("GET /users?search", function () {
+    let jwt;
+    let userId = "";
+
+    before(async function () {
+      userId = await addUser();
+      jwt = authService.generateAuthToken({ userId });
+
+      await addOrUpdate(userSearchParamsData[0]);
+      await addOrUpdate(userSearchParamsData[1]);
+    });
+
+    after(async function () {
+      await cleanDb();
+    });
+
+    it(`Test Scenario ${i}: ${item.desc}`, function (done) {
+      chai
+        .request(app)
+        .get("/users")
+        .query({ search: item.value })
+        .set("cookie", `${cookieName}=${jwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+          if (item.value === "mu") {
+            expect(res).to.have.status(200);
+            expect(res.body).to.be.a("object");
+            expect(res.body.message).to.equal("No users are present");
+          } else if (
+            item.value === "an" ||
+            item.value === "AN" ||
+            typeof item.value === "number" ||
+            item.value === ""
+          ) {
+            expect(res).to.have.status(200);
+            expect(res.body).to.be.a("object");
+            expect(res.body.message).to.equal("Users returned successfully!");
+            expect(res.body.users).to.be.a("array");
+          }
+
```

Comment:
I think we are complicating things here by running it in the loop
Like conditional branching for each test case, 
This should be separately handled

Reply:
okay will write individual tests.

Reply:
done